replications/internal/queue_management.go

```diff
@@ -143,3 +146,35 @@ func (qm *durableQueueManager) CurrentQueueSizes(ids []platform.ID) (map[platfor
 
 	return sizes, nil
 }
+
+// EnqueuePoints persists a set of points to a replication's durable queue.
+func (qm *durableQueueManager) EnqueuePoints(replicationID platform.ID, points []models.Point) error {
+	qm.mutex.RLock()
+	defer qm.mutex.RUnlock()
+
+	if qm.replicationQueues[replicationID] == nil {
+		return fmt.Errorf("durable queue not found for replication ID %q", replicationID)
+	}
+
+	// Serialize the points back into line protocol.
+	// gzip the data so it takes up less room on disk. We can send the gzip directly to the remote
+	// without needing to decompress it.
+	var buf bytes.Buffer
```

Comment:
This will will be duplicated `N` times for all `N` queues associated with a local bucket... would be nice to move it out of this method so we only need to do it once.

Reply:
Great idea getting both serialization and gzip into a single execution.